docs/DeveloperGuide.md

```diff
@@ -245,10 +245,12 @@ gradlew run -Dargs="-c ./configs/ -o output_path/ -s 21/10/2017 -u 21/11/2017 -f
 ## HTML Report
 The source files for the report is located in [`frontend/src`](../frontend/src) and is built by [spuild](https://github.com/ongspxm/spuild2) before being packaged into the JAR file to be extracted as part of the report.
 
-The main HTML file is generated from [`frontend/src/index.jade`](../frontend/src/index.jade).
+The main HTML file is generated from [`frontend/src/index.pug`](../frontend/src/index.pug).
 
-[Vue](https://vuejs.org/v2/api/) (pronounced /vjuː/, like view) is a progressive framework for building user interfaces. It is heavily utilized in the report to dynamically update the information in the various views. (Style guide available [here](https://vuejs.org/v2/style-guide/), Developer tool available [here](https://chrome.google.com/webstore/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd)).
+[Vue](https://vuejs.org/v2/api/) (pronounced /vjuː/, like view) is a progressive framework for building user interfaces. It is heavily utilized in the report to dynamically update the information in the various views. (Style guide available [here](https://vuejs.org/v2/style-guide/), Developer tool available [here](https://chrome.google.com/webstore/detail/vuejs-devtools/nhdogjmejiglipccpnnnanhbledajbpd)). Vue lifecycle hooks are the defined methods which gets executed in a certain stage of the Vue object lifespan. The following is the lifecycle diagram indicating the hook sequence:
```

Comment:
In docs/images/vue-lifecycle-diagram.png :

Might not be a good idea to split into two halves. Either draw a dotted line to link the arrows together, paste the image from https://vuejs.org/images/lifecycle.png as a whole, or link to [this page](https://vuejs.org/v2/guide/instance.html#Lifecycle-Diagram) instead for them to read up on.

Also adding on to @bluein-green , if pasting the image file, do ensure that they allow modification and/or reproduction of the image file.